### PR TITLE
Remove 1x1 matmul fallback on convolution and generalize convo…

### PR DESCRIPTION
**Pull request by tapspatel**
…lution kernel. Remove unused convolution kernels

### CODE REVIEW

**Comment by tapspatel** [comment]
Issue: https://github.com/tenstorrent/tt-metal/issues/8407

File: tt_eager/tt_dnn/op_library/conv/kernels/reader_conv_activations_2d_mcast_padded_with_halo_3x3_weights_v2.cpp [hunk 1]
```diff
@@ -49,7 +49,10 @@ void kernel_main() {
     constexpr uint32_t window_outer = get_compile_time_arg_val(6);
     constexpr uint32_t window_inner = get_compile_time_arg_val(7);
     constexpr uint32_t act_block_h_datums = get_compile_time_arg_val(8);
-
+    constexpr uint32_t in0_block_num_tiles = get_compile_time_arg_val(9);
+    constexpr uint32_t weight_size_w = get_compile_time_arg_val(10);
+    constexpr uint32_t act_block_w_extra_align_bytes = get_compile_time_arg_val(12);
+    constexpr uint32_t weight_size_h = get_compile_time_arg_val(13);
```
Comment (TT-BrianLiu) [inline tt_eager/tt_dnn/op_library/conv/kernels/reader_conv_activations_2d_mcast_padded_with_halo_3x3_weights_v2.cpp:55]: what happened to 11?

**Review by TT-BrianLiu — APPROVED** [review]
Minor comment

**Comment by mywoodstock** [comment]
@tt-nshanker could you please scan through to see none of the needed kernels are deleted?

**Review by mywoodstock — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/conv/multi_core_optimized_conv_sharded/optimized_conv_op_sharded_v2.cpp [hunk 2]
```diff
@@ -719,9 +721,16 @@ operation::ProgramWithCallbacks multi_core_optimized_conv_sharded_v2_impl(
         // If conv_act_c_blocks > 1 and we have 2D conv with sharded input, we always read entire 3x3 window before
         // pushing in reader/writer
         // TODO: Generalize this to not make this assumption
-        read_3x3_window_in_inner_loop = true;
+        read_window_in_inner_loop = true;
         num_weight_cb_tiles *= weight_size_h * weight_size_w;
         num_act_cb_tiles *= weight_size_h * weight_size_w;
+    } else if (a.memory_config().is_sharded() and weight_size_h == 1 and weight_size_w == 1 and stride_h == 2 and weight_width_sliced) {
```
Comment (tt-nshanker) [inline tt_eager/tt_dnn/op_library/conv/multi_core_optimized_conv_sharded/optimized_conv_op_sharded_v2.cpp:727]: the code for the prior if condition and this else if can be merged into one

**Review by cfjchu — APPROVED** [review]

**Review by xanderchin — APPROVED** [review]

**Review by tt-nshanker — APPROVED** [review]

**Pull request merged**